Assets/GameAssets/Player/PlayerScripts/PlayerInteraction.cs

```diff
@@ -21,11 +21,22 @@ public class PlayerInteraction : MonoBehaviour
     [SerializeField] private float _maxReach;
     [Tooltip("Ray that is cast from camera to find interactable objects")]
     private Ray _ray;
+    [Tooltip("UI object that will be toggled when you can or cannot interact with an object")]
+    private GameObject _interactUI;
 
     //Input
     private PlayerInput _playerInput;
     private InputAction _interactInput;
 
+    /// <summary>
+    /// Right away finds the InteractionUI object for future use and disables it before player can see
+    /// </summary>
+    private void Awake()
+    {
+        _interactUI = GameObject.Find("InteractionUI");
```

Comment:
I would prefer the UI code to be in its own script and can listen for an event that happens when the player looks at and away from the interactable

Reply:
Especially in the event that the Interaction UI code eventually will need to be more complicated than a simple enable and disable object, having a system in place that handles an event when the player looks at an interactable would be beneficial in the long run